src/flowgger/fuzzer.rs

```diff
@@ -0,0 +1,210 @@
+        }
+
+        drop(sender_guard);
+    }
+
+    /// Start an input listener which writes data to the output file once received.
+    pub fn start_file_output(config: &Config, rx: Receiver<Vec<u8>>){
+
+        let output_format = config
+            .lookup("output.format")
+            .map_or(DEFAULT_OUTPUT_FORMAT, |x| {
+                x.as_str().expect("output.format must be a string")
+            });
+
+        let output = get_file_output(&config);
+        let output_type = config
+            .lookup("output.type")
+            .map_or(DEFAULT_OUTPUT_TYPE, |x| {
+                x.as_str().expect("output.type must be a string")
+            });
+
+        let _output_framing = match config.lookup("output.framing") {
+            Some(framing) => framing.as_str().expect("output.framing must be a string"),
+            None => match (output_format, output_type) {
+                ("capnp", _) | (_, "kafka") => "noop",
+                (_, "debug") | ("ltsv", _) => "line",
+                ("gelf", _) => "nul",
+                _ => DEFAULT_OUTPUT_FRAMING,
+            },
+        };
+        let merger: Option<Box<dyn Merger>> = Some(Box::new(LineMerger::new(&config)) as Box<dyn Merger>);
+
+        let arx = Arc::new(Mutex::new(rx));
+        output.start(arx, merger);
+    }
+
+    fn get_file_output(config: &Config) -> Box<dyn Output> {
+        Box::new(FileOutput::new(config)) as Box<dyn Output>
+    }
+    
+    pub fn fuzz_target_rfc3164(data: String) {
+        let config = STATIC_CONFIG.clone();
+        
+        let mut sender_guard = match SYNC_SENDER.lock() {
+            Ok(guard) => guard,
+            Err(_poisoned_error) => {
+                // Handle the poisoned Mutex
+                let guard = _poisoned_error.into_inner();
+                guard
+            }
+        };
+        let tx: &mut SyncSender<Vec<u8>> = sender_guard.as_mut().unwrap();
+
+        let encoder = get_encoder_rfc3164(&config);
+        let decoder = get_decoder_rfc3164(&config);
+        let (decoder, encoder): (Box<dyn Decoder>, Box<dyn Encoder>) =
+            (decoder.clone_boxed(), encoder.clone_boxed());
+        let _result = handle_record_maybe_compressed(data.as_bytes(), &tx, &decoder, &encoder);
+    }
+
+    // Check for the result
+    // On invalid input, no logs are expected to be written to the output file
+    // For valid inputs, analyze each log entry, check that the hostname and appnames are defined
+    fn check_result(file_output_path: &str)-> Result<(), std::io::Error> {
+
+        let mut sender_guard = match SYNC_SENDER.lock() {
+            Ok(guard) => guard,
+            Err(_poisoned_error) => {
+                // Handle the poisoned Mutex
+                let guard = _poisoned_error.into_inner();
+                guard
+            }
+        };
+        let tx: SyncSender<Vec<u8>> = sender_guard.take().unwrap();
+        drop(tx);
+
+        thread::sleep(time::Duration::from_millis(1000));
```

Comment:
explain why you need to sleep please. if the drop is not blocking and you're waiting, how do you know 1 sec is enough ?